### PR TITLE
nlohmann_json: 3.9.1 -> 3.10.0

### PR DESCRIPTION
**Pull request by sternenseemann**
* Patch has been merged in the mean time

* Test issue supposedly fixed upstream

https://github.com/nlohmann/json/releases/tag/v3.10.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
@ofborg build hydra-unstable nix nixStatic nixUnstable

File: pkgs/development/libraries/nlohmann_json/default.nix [hunk 1]
```diff
@@ -1,35 +1,39 @@
...
+    "-DJSON_TestDataDirectory=${testData}"
   ];
 
-  # A test cause the build to timeout https://github.com/nlohmann/json/issues/1816
-  #doCheck = stdenv.hostPlatform == stdenv.buildPlatform;
-  doCheck = false;
+  # skip tests that require git or modify “installed files”
+  preCheck = ''
+    checkFlagsArray+=("ARGS=-LE 'not_reproducible|git_required'")
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/nlohmann_json/default.nix:35]:
```suggestion
  checkFlags = [
    "ARGS=-LE 'not_reproducible|git_required'"
  ];
```

Comment (sternenseemann) [reply pkgs/development/libraries/nlohmann_json/default.nix:35]: Use of `checkFlagsArray` is deliberate — otherwise the arguments are split at the space.

**Comment by sternenseemann** [comment]
Getting these failures, haven't checked yet if they are related:

```
nixos-install-tools python38Packages.qiskit python38Packages.qiskit-aqua python39Packages.qiskit python39Packages.qiskit-aqua
```

**Pull request closed without merging**